rclpy/rclpy/time_source.py

```diff
@@ -119,9 +127,19 @@ def _on_parameter_event(self, parameter_list):
                 if parameter.type_ == Parameter.Type.BOOL:
                     self.ros_time_is_active = parameter.value
                 else:
-                    self._node.get_logger().error(
-                        '{} parameter set to something besides a bool'
-                        .format(USE_SIM_TIME_NAME))
+                    node = self._get_node()
+                    if node:
+                        node.get_logger().error(
+                            '{} parameter set to something besides a bool'
+                            .format(USE_SIM_TIME_NAME))
                 break
 
         return SetParametersResult(successful=True)
+
+    def _get_node(self):
+        if self._node_weak_ref is not None:
+            node = self._node_weak_ref()
+            if node is None:
+                raise RuntimeError('Attached node has gone out of scope before expected')
```

Comment:
"before expected" sounds weird to me (I am not a native English speaker though). Maybe:

> Attached node has gone out of scope already

But actually in all the calls you check if the returned value is `None`. In which cases is the weak ref still set but the node is already out of scope and you actually want an exception here instead of just returning `None`?

Reply:
Yeah, returning just `None` is also an option. I updated the code.